2006-04-03	Nicolas Setton  <setton@adacore.com>

	* gtk/gtkpixmap.c (gtk_pixmap_set): Do not check the depth of a
	null drawable.
	Backport patch already present in the Gtk+ CVS.
	For F324-013.

2006-03-23	Nicolas Setton  <setton@adacore.com>
	
	* gtk/gtktextview.c (gtk_text_view_value_changed): Do not invalidate
	the window if the view is not realized.
	Submitted at http://bugzilla.gnome.org/show_bug.cgi?id=335717.
	Patch has been applied in the Gtk+ CVS.
        For F323-006.

2006-02-23	Arnaud Charlet	<charlet@adacore.com>

	* gdk/x11/gdkasync.c: #define ANSICPP on Solaris, to work around
	bug in X11/Xmd.h on Solaris 8.

2006-01-04	Arnaud Charlet	<charlet@adacore.com>

	* gtk/gtktextview.c (gtk_text_view_start_selection_dnd): Disabled,
	since not functional.
	For C928-003.

2003-11-27	Arnaud Charlet	<charlet@adacore.com>

	* gtk/gtktextview.c (incremental_validate_callback): Increase amount
	of characters validated at each call, this gives a better behavior
	when loading big files.
	Submitted at http://bugzilla.gnome.org/show_bug.cgi?id=332057

--- gtk/gtktextview.c.old	2006-02-17 16:11:04.000000000 +0100
+++ gtk/gtktextview.c	2006-02-20 17:08:54.691581240 +0100
@@ -3273,7 +3273,7 @@ incremental_validate_callback (gpointer 
   
   DV(g_print(G_STRLOC"\n"));
   
-  gtk_text_layout_validate (text_view->layout, 2000);
+  gtk_text_layout_validate (text_view->layout, 20000);
 
   gtk_text_view_update_adjustments (text_view);
   
@@ -6028,6 +6028,8 @@ gtk_text_view_start_selection_dnd (GtkTe
   GdkDragContext *context;
   GtkTargetList  *target_list;
 
+  return;
+
   text_view->drag_start_x = -1;
   text_view->drag_start_y = -1;
   text_view->pending_place_cursor_button = 0;
--- gdk/x11/gdkasync.c.old	Thu Feb 23 11:30:46 2006
+++ gdk/x11/gdkasync.c	Thu Feb 23 11:48:14 2006
@@ -47,6 +47,11 @@ in this Software without prior written a
 #ifdef NEED_XIPROTO_H_FOR_XREPLY
 #include <X11/extensions/XIproto.h>
 #endif
+
+#if defined (sun) && defined (__SVR4)
+#define ANSICPP
+#endif
+
 #include <X11/Xlibint.h>
 #include "gdkasync.h"
 #include "gdkx.h"
*** gtk/gtktextview.c.original	Thu Mar 23 18:53:45 2006
--- gtk/gtktextview.c	Thu Mar 23 18:56:11 2006
*************** gtk_text_view_value_changed (GtkAdjustme
*** 6570,6576 ****
         */
        if (text_view->width_changed)
  	{
! 	  gdk_window_invalidate_rect (text_view->text_window->bin_window, NULL, FALSE);
  	  
  	  text_view->width_changed = FALSE;
  	}
--- 6570,6578 ----
         */
        if (text_view->width_changed)
  	{
! 	  /* No need to invalidate the window if the view is not realized.  */
! 	  if (text_view->text_window->bin_window)
! 	    gdk_window_invalidate_rect (text_view->text_window->bin_window, NULL, FALSE);
  	  
  	  text_view->width_changed = FALSE;
  	}
*** gtk/gtkpixmap.c.original	Mon Apr  3 12:44:33 2006
--- gtk/gtkpixmap.c	Mon Apr  3 12:45:03 2006
*************** gtk_pixmap_set (GtkPixmap *pixmap,
*** 131,137 ****
    gint oldheight;
  
    g_return_if_fail (GTK_IS_PIXMAP (pixmap));
!   g_return_if_fail (gdk_colormap_get_visual (gtk_widget_get_colormap (GTK_WIDGET (pixmap)))->depth == gdk_drawable_get_depth (GDK_DRAWABLE (val)));
  
    if (pixmap->pixmap != val)
      {
--- 131,138 ----
    gint oldheight;
  
    g_return_if_fail (GTK_IS_PIXMAP (pixmap));
!   if(GDK_IS_DRAWABLE(val))
!     g_return_if_fail (gdk_colormap_get_visual (gtk_widget_get_colormap (GTK_WIDGET (pixmap)))->depth == gdk_drawable_get_depth (GDK_DRAWABLE (val)));
  
    if (pixmap->pixmap != val)
      {
